testsuite/tests/lib-bigarray/weak_bigarray.ml

```diff
@@ -0,0 +1,27 @@
+
+
+(** check that custom block are not copied by Weak.get_copy *)
+
+open Bigarray
+open Bigarray.Array1
+
+let () =
+  let a = ref (create float64 c_layout 10) in
+  Gc.compact ();
+  set !a 0 42.;
+
+  let w = Weak.create 1 in
+  Weak.set w 0 (Some !a);
+
+  let b =
+    match Weak.get_copy w 0 with
+    | None -> assert false
+    | Some b -> b
+  in
+  Printf.printf "a.(0) = %f\n" (get !a 0);
+  Printf.printf "b.(0) = %f\n" (get b 0);
+  a := create float64 c_layout 10;
+  Gc.compact ();
+
+  let c = create float64 c_layout 10 in
```

Comment:
I just re-tested this with a fresh install of `4.03`, and I get the same behavior with or without the `let c = ...` line (modulo the warning). That is:
```
 $ ocamlbuild -pkg bigarray t.native --
Finished, 4 targets (0 cached) in 00:00:00.
a.(0) = 42.000000
b.(0) = 42.000000
b.(0) = 0.000000
```